### PR TITLE
Filter ingress class resources by name

### LINKED ISSUE #7896: Conflict: traefik instances with ingressClass
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

2 separate traefik instances and one ingress object using IngressClassName

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: helloworld
  namespace: test
  annotations:
#    kubernetes.io/ingress.class: traefik # can't set if ingressClassName is set
spec:
  ingressClassName: traefik-external
  rules:
  - host: test.mydomain.com
  # ...
```

```yaml
apiVersion: networking.k8s.io/v1
kind: IngressClass
metadata:
  annotations:
    ingressclass.kubernetes.io/is-default-class: "false"
  name: traefik-external
  spec:
    controller: traefik.io/ingress-controller
```
  
traefik-external cli parameters:
```
    --providers.kubernetesingress
    --providers.kubernetescrd.ingressclass=traefik-external
    --providers.kubernetesingress.ingressclass=traefik-external
```

**Result: This works fine at all times!**

Now I create **additionally** a new traefik instance "traefik-internal" which is exactly the same except for the ingressClass name:

```yaml
apiVersion: networking.k8s.io/v1
kind: IngressClass
metadata:
  annotations:
    ingressclass.kubernetes.io/is-default-class: "false"
  name: traefik-internal
  spec:
    controller: traefik.io/ingress-controller
```

traefik-internal cli parameters:
```
    --providers.kubernetesingress
    --providers.kubernetescrd.ingressclass=traefik-internal
    --providers.kubernetesingress.ingressclass=traefik-internal
```

### What did you expect to see?

My helloworld should be served by the `traefik-external` instance as before. The `traefik-internal` should not serve this ingress object.

### What did you see instead?

What I am seeing now is that randomly about every 5-15 seconds the http service `test-helloworld-80@kubernetes` and the corresponding http route appears and disappears again.
I can see this on both traefik dashboards, the traefik-external and the traefik-internal one.

### Output of `traefik version`: (_What version of Traefik are you using?_)

traefik 2.4.0

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

kubernetes 1.19

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

every 5-10 seconds:
```
level=debug msg="Configuration received from provider kubernetes: {\"http\":{\"routers\":{\"helloworld-test-test-mydomain-com\":{\"service\":\"test-helloworld-80\",\"rule\":\"Host(`test.mydomain.com`) \\u0026\\u0026 PathPrefix(`/`)\"}},\"services\":{\"test-helloworld-80\":{\"loadBalancer\":{\"servers\":[{\"url\":\"http://10.42.0.43:80\"}],\"passHostHeader\":true}}}},\"tcp\":{},\"tls\":{}}" providerName=kubernetes
level=debug msg="Creating middleware" entryPointName=web routerName=helloworld-test-test-mydomain-com@kubernetes serviceName=test-helloworld-80 middlewareName=pipelining middlewareType=Pipelining
level=debug msg="Creating load-balancer" routerName=helloworld-test-test-mydomain-com@kubernetes serviceName=test-helloworld-80 entryPointName=web
level=debug msg="Creating server 0 http://10.42.0.43:80" entryPointName=web routerName=helloworld-test-test-mydomain-com@kubernetes serviceName=test-helloworld-80 serverName=0
level=debug msg="Added outgoing tracing middleware test-helloworld-80" routerName=helloworld-test-test-mydomain-com@kubernetes middlewareName=tracing middlewareType=TracingForwarder entryPointName=web
```

### PR DESCRIPTION
**Pull request by tomMoulard**
### What does this PR do?

This PR fixes #7896.

### Motivation

Be able to distinguish different ingressclass resource by name.
 
### More

- [X] Added/updated tests
- [X] Added/updated documentation

### Additional Notes

Co-authored-by: SantoDE <manuel.zapf@traefik.io>

### CODE REVIEW

**Review by jbdoumenjou — APPROVED** [review]
LGTM :+1:

File: pkg/provider/kubernetes/ingress/client.go [hunk 1]
```diff
@@ -476,3 +476,15 @@ func supportsIngressClass(serverVersion *version.Version) bool {
...
+// filterIngressClassByName return a slice containing ingressclasses with the correct name.
+func filterIngressClassByName(ingressClassName string, ics []*networkingv1beta1.IngressClass) []*networkingv1beta1.IngressClass {
+	var ingressClasses []*networkingv1beta1.IngressClass
+
+	for _, ic := range ics {
+		if ic.Name == ingressClassName {
+			ingressClasses = append(ingressClasses, ic)
+		}
+	}
+	return ingressClasses
```
Comment (rtribotte) [inline pkg/provider/kubernetes/ingress/client.go:489]:
```suggestion
	}
	
	return ingressClasses
```

**Review by rtribotte — APPROVED** [review]
LGTM 👍

**Review by ldez — APPROVED** [review]
Thanks :+1:

**Pull request merged**